compiler/hash-exhaustiveness/src/deconstruct.rs

```diff
@@ -214,7 +214,7 @@ impl<E: ExhaustivenessEnv> fmt::Debug for ExhaustivenessFmtCtx<'_, Deconstructed
                         // currently active.
                         if let DeconstructedCtor::Variant(index) = ctor {
                             let ctors = data_def.borrow().ctors.assert_defined();
-                            let ctor_name = CtorDefId(ctors.elements(), *index).borrow().name;
+                            let ctor_name = CtorDefId::new(ctors.elements(), *index).borrow().name;
```

Comment:
Maybe we can provide a method on sequence stores `from_raw_parts` that actually does the assertion that the index is within bounds?

Reply:
We could do a `debug_assert!()` in the `new()` constructor?